.gitlab-ci.yml

```diff
@@ -27,58 +27,52 @@ builders-and-yarn:
   stage: builders
   image: registry.gitlab.com/magda-data/magda/data61/magda-builder-docker:master
   retry: 1
+  before_script:
+    - yarn install --no-progress --cache-folder .yarn
   cache:
-    key: $CI_JOB_NAME-$CACHE_VERSION
+    key: JS-$JS_CACHE_VERSION
     paths:
-      - .yarn
-  before_script:
-    - yarn install --no-progress --child-concurrency 1 --cache-folder .yarn
+      - $CI_PROJECT_DIR/.yarn
+      - $CI_PROJECT_DIR/node_modules/
+      - "$CI_PROJECT_DIR/*/node_modules"
   services:
     - docker:dind
   script:
     - docker login -u gitlab-ci-token -p $CI_JOB_TOKEN $CI_REGISTRY
     - yarn run in-submodules -- -f categories.builder=true -- run docker-build-prod --include-filtered-dependencies -- -- --repository=$CI_REGISTRY/magda-data/magda --version=$CI_COMMIT_REF_SLUG --cacheFromVersion=master
-  artifacts:
-    paths:
-      - "node_modules"
-      - "*/node_modules"
-      - "packages/*/node_modules"
-      - "yarn.lock"
-    expire_in: 7 days
+  tags:
+    - heavy
 
-# Make sure sbt depenencies, plugins are in place, cached (only for this job) and pass to following stage as artifacts
-sbt-prebuild:
-  stage: sbt-prebuild
+check-scala-formatting:
+  stage: prebuild
   image: registry.gitlab.com/magda-data/magda/data61/magda-builder-scala:$CI_COMMIT_REF_SLUG
   cache:
-    key: $CI_JOB_NAME-$CACHE_VERSION
+    key: SCALA-$SCALA_CACHE_VERSION
     paths:
       - "$CI_PROJECT_DIR/sbt-cache"
+      - "$CI_PROJECT_DIR/target"
```

Comment:
Probably we still want to use `artifacts`  to pass intermediate build results between stages / jobs rather than cache (as suggested [here](https://docs.gitlab.com/ee/ci/caching/))?
If we passing scala `target` folder which contains build result, we might having issue that different commit share different build result between jobs in the same pipeline.
e.g. job step 2 may build on build result from (older / other commit) upstream job step1